src/connector/src/source/nats/source/message.rs

```diff
@@ -20,12 +20,12 @@ use crate::source::base::SourceMessage;
 use crate::source::SourceMeta;
 
 impl SourceMessage {
-    pub fn from_nats_message(message: async_nats::Message) -> Self {
+    pub fn from_nats_jetstream_message(message: async_nats::jetstream::message::Message) -> Self {
         SourceMessage {
             key: None,
-            payload: Some(message.payload.to_vec()),
-            // Nats message doesn't have offset
-            offset: "".to_string(),
+            payload: Some(message.message.payload.to_vec()),
+            // For nats jetstream, use sequence id as offset
+            offset: message.info().unwrap().consumer_sequence.to_string(),
```

Comment:
Does the offset number grow one by one in a subject?

Reply:
I do a experiments, and find the subquence number will always grow one by one in a subject.

Reply:
After checked the docs, I guess the reason for second case is forget the ack the message and set the ack policy for consumer.

Reply:
> But one message not equal to one sequence number.
For example: send msg id 0..9, then rw recieve sequence number from 0..9. Then I send msg id 10..20, the rw may received msg id from 10..25. The final data in table is correct. But rw may get more messages then I send. Not sure whether result by nats design.

In this example, without double_ack, will the consumer receive messages from [21, ...) if the consumer resumes consumption on msg id 25?

Reply:
bohan recommand to use None ack policy for consumer. Because we don't depend on the information from server. If use none ack, the number will be matched.